samples/containers/poetry/src/main/java/com/squareup/sample/poetry/PoemsBrowserWorkflow.kt

```diff
@@ -2,54 +2,219 @@ package com.squareup.sample.poetry
 
 import com.squareup.sample.container.overviewdetail.OverviewDetailScreen
 import com.squareup.sample.poetry.PoemListRendering.Companion.NO_POEM_SELECTED
+import com.squareup.sample.poetry.PoemWorkflow.Output.ClosePoem
+import com.squareup.sample.poetry.PoemWorkflow.Output.Loading
+import com.squareup.sample.poetry.PoemWorkflow.PoemRenderingWithLoading
+import com.squareup.sample.poetry.PoemsBrowserWorkflow.Config
+import com.squareup.sample.poetry.PoemsBrowserWorkflow.State
+import com.squareup.sample.poetry.PoemsBrowserWorkflow.State.ComplexCall
+import com.squareup.sample.poetry.PoemsBrowserWorkflow.State.Initializing
+import com.squareup.sample.poetry.PoemsBrowserWorkflow.State.NoSelection
+import com.squareup.sample.poetry.PoemsBrowserWorkflow.State.Selected
 import com.squareup.sample.poetry.model.Poem
+import com.squareup.sample.poetry.views.BlankScreen
+import com.squareup.sample.poetry.views.LoaderSpinner
+import com.squareup.sample.poetry.views.MayBeLoadingScreen
 import com.squareup.workflow1.Snapshot
 import com.squareup.workflow1.StatefulWorkflow
+import com.squareup.workflow1.Worker
 import com.squareup.workflow1.action
 import com.squareup.workflow1.parse
+import com.squareup.workflow1.runningWorker
 import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
 import com.squareup.workflow1.ui.backstack.BackStackScreen
-
-typealias SelectedPoem = Int
+import kotlinx.coroutines.delay
 
 object PoemsBrowserWorkflow :
-  StatefulWorkflow<List<Poem>, SelectedPoem, Nothing, OverviewDetailScreen>() {
+  StatefulWorkflow<Config, State, Nothing, MayBeLoadingScreen>() {
+
+  data class Config(
+    val poems: List<Poem>,
+    val simulatedPerfConfig: SimulatedPerfConfig = SimulatedPerfConfig.NO_SIMULATED_PERF
+  )
+
+  sealed class State {
+    var poemLoading = false
+
+    // N.B. This state is a smell. We include it as possible in order to be able to config similar
+    // to real-world usage. It should be folded into [NoSelection] at the very least.
+    object Initializing : State()
+    data class ComplexCall(
+      val payload: Int
+    ) : State()
+
+    object NoSelection : State()
+    data class Selected(val poemIndex: Int) : State()
+  }
 
   override fun initialState(
-    props: List<Poem>,
+    props: Config,
     snapshot: Snapshot?
-  ): SelectedPoem {
-    return snapshot?.bytes?.parse { source -> source.readInt() }
-      ?: NO_POEM_SELECTED
+  ): State {
```

Comment:
If you do choose to go the DI route, the benchmark implementation could just return a null `Snapshot`.